src/RTTI/RTTIUtil.hpp

```diff
@@ -37,11 +38,32 @@
  * note that there is not semicolon after the makro!
  * Futhermore, the makro invocation needs to be in `public`-land of the component.
  */
-#define REGOTH_DECLARE_RTTI(classname)                           \
-  friend class bs::SceneObject;                                  \
+#define REGOTH_DECLARE_RTTI(classname)      \
+  friend class bs::SceneObject;             \
+  friend class RTTI_##classname;            \
+  static bs::RTTITypeBase* getRTTIStatic(); \
+  decltype(classname::getRTTIStatic()) getRTTI() const override;
+
+/**
+ * See REGOTH_DECLARE_RTTI. This is a special version for IReflectables,
+ * which cannot be default constructed by default. Should be used in favor
+ * of the plain REGOTH_DECLARE_RTTI if used for IReflectables, except for
+ * abstract classes.
+ *
+ * @note rttiCreateEmpty() is private so sub-classes which forget to use this
+ * makro wont have access to the wrong rttiCreateEmpty().
+ **/
+#define REGOTH_DECLARE_RTTI_FOR_REFLECTABLE(classname)           \
+public:                                                          \
   friend class RTTI_##classname;                                 \
   static bs::RTTITypeBase* getRTTIStatic();                      \
-  decltype(classname::getRTTIStatic()) getRTTI() const override; 
+  decltype(classname::getRTTIStatic()) getRTTI() const override; \
+                                                                 \
+private:                                                         \
+  static bs::SPtr<classname> rttiCreateEmpty()                   \
```

Comment:
Better make this inline, otherwise it could (?) violate the one definition rule.

Reply:
The linker would complain if you were to define the same static function twice. Do you have an example on what could go wrong?

Reply:
I was just wondering if anything could go wrong by *defining* a non-inline member function in the header and the header being included in multiple cpps. But probably not, as the compiler doesn't seem to care in a minimal example.

Reply:
Ah, now I know what you mean. For classes this is indeed a bit weird, I agree.